### PR TITLE
Snow machine config validation webhook for static validations

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
#2669 

*Description of changes:*

Validating the Snow Machine Config via the validating webhook.

Testing (if applicable):

- Create docker cluster with tilt and attach a snow machine config on the cluster with specific values set. Check once attached if the values are validated.
- Unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (g-gaston) [inline pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go:1]: Curious why this is changing

What commands did you run?
What go version are you using?

File: test/e2e/vsphere_autoimport_test.go [hunk 2]
```diff
@@ -163,4 +163,4 @@ func TestVSphereKubernetes123BottlerocketAutoimport(t *testing.T) {
 		framework.WithEnvVar(features.K8s123SupportEnvVar, "true"),
 	)
 	runAutoImportFlow(test, provider)
-}
\ No newline at end of file
+}
```
Comment (g-gaston) [inline test/e2e/vsphere_autoimport_test.go:166]: ??

Comment (mitalipaygude) [reply test/e2e/vsphere_autoimport_test.go:166]: gofumpt added this

File: pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go (hunk 1 above)
Comment (g-gaston) [reply pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go:1]: we shouldn't gofumpt autogenerated files, could we remove it?

File: test/e2e/vsphere_autoimport_test.go (hunk 2 above)
Comment (g-gaston) [reply test/e2e/vsphere_autoimport_test.go:166]: I'm fine if you want to change this, but it should be in a different PR, this one is not even touching that package

Plus, if the linter is not complaining, we shouldn't reformat random files

File: pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go (hunk 1 above)
Comment (mitalipaygude) [reply pkg/api/v1alpha1/thirdparty/tinkerbell/zz_generated.deepcopy.go:1]: make was failing and asking me to reformat and then it reformatted all these files. let me remove them.

**Review by g-gaston — APPROVED** [review]

**Pull request merged**